Web Inspector: Screenshots timeline is unsupported in WebKitLegacy, we should disable it when inspect such targets

patrickangle opened the pull request:
#### 2129556ec693c9909c60b6576f8a1c08c9f03a4d
<pre>
Web Inspector: Screenshots timeline is unsupported in WebKitLegacy, we should disable it when inspect such targets
<a href="https://bugs.webkit.org/show_bug.cgi?id=244301">https://bugs.webkit.org/show_bug.cgi?id=244301</a>
rdar://problem/99094864

Reviewed by Devin Rousso and BJ Burg.

Screenshots are taken when a composite occurs to populate the screenshots timeline. Composites do not occurs for
the same types of updates in WebKitLegacy as they do in WebKit2. Due to this, the timeline currently appears
broken on WebKitLegacy targets. Until we can do the work in webkit.org/b/251113 to support WebKitLegacy, we
instead will disable the Screenshots timeline when inspecting WebKitLegacy targets (the &quot;Page&quot; debuggable type,
as opposed to the WK2 &quot;Web Page&quot; debuggable type).

* LayoutTests/platform/mac-wk1/TestExpectations:
* Source/WebInspectorUI/UserInterface/Models/ScreenshotsInstrument.js:
(WI.ScreenshotsInstrument.supported):

Canonical link: <a href="https://commits.webkit.org/259326@main">https://commits.webkit.org/259326@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a9a9ab22fa7d4a486b0f1600b9330f6fff6cc9db

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104638 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13712 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/37539 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/113913 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/174128 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/108555 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/14837 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4643 "Built successfully") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/96968 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/112853 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/110401 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/11442 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/37539 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/96968 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/108110 "Passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/93298 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/37539 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/96968 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/1/builds/94596 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/7069 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/37539 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/92530 "Built successfully") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/83/builds/4838 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/7177 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/4026 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/46/builds/30101 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🧪 services~~](https://ews-build.webkit.org/#/builders/20/builds/103467 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/13224 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/37539 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/101216 "Built successfully") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/74/builds/6433 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/79/builds/8966 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🧪 jsc-mips-tests~~](https://ews-build.webkit.org/#/builders/45/builds/25168 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on LayoutTests/platform/mac-wk1/TestExpectations:
```diff
@@ -960,6 +960,9 @@ media/now-playing-status-without-media.html [ Skip ]
 http/tests/media/video-webm-stall.html
 
+# Screenshots timeline does not capture screenshots for changes in WebKitLegacy.
+webkit.org/b/251113 inspector/timeline/timeline-event-screenshots.html [ Failure ]
```
why not `Skip`?

dcrousso commented on Source/WebInspectorUI/UserInterface/Models/ScreenshotsInstrument.js:
```diff
@@ -36,6 +36,10 @@ WI.ScreenshotsInstrument = class ScreenshotsInstrument extends WI.Instrument
     static supported()
     {
+        // FIXME: <webkit.org/b/251113> Screenshots timeline does not capture screenshots for changes in WebKitLegacy.
+        if (WI.sharedApp.debuggableType === WI.DebuggableType.Page)
```
Will this work for iOS 13.0 and earlier where we have to wait for a `Target.exists` command to fail before "realizing" that we're dealing with WKL?

NIT: I wonder if we should create a `WI.AppControllerBase.prototype.get isWebKitLegacy` (similar to the `WI.AppControllerBase.prototype.get isWebDebuggable` we already have)

patrickangle replied:
Not sure about pre-iOS 13. I'll take a look. But the check below this one will fail pre-iOS 16.0. But if we abstract this logic out I'll have to double-check pre-iOS 13 compatibility.

dcrousso replied:
oh lol good point i forgot about the other check 🤣

patrickangle committed:
Web Inspector: Screenshots timeline is unsupported in WebKitLegacy, we should disable it when inspect such targets
https://bugs.webkit.org/show_bug.cgi?id=244301
rdar://problem/99094864

Reviewed by Devin Rousso and BJ Burg.

Screenshots are taken when a composite occurs to populate the screenshots timeline. Composites do not occurs for
the same types of updates in WebKitLegacy as they do in WebKit2. Due to this, the timeline currently appears
broken on WebKitLegacy targets. Until we can do the work in webkit.org/b/251113 to support WebKitLegacy, we
instead will disable the Screenshots timeline when inspecting WebKitLegacy targets (the "Page" debuggable type,
as opposed to the WK2 "Web Page" debuggable type).

* LayoutTests/platform/mac-wk1/TestExpectations:
* Source/WebInspectorUI/UserInterface/Models/ScreenshotsInstrument.js:
(WI.ScreenshotsInstrument.supported):

Canonical link: https://commits.webkit.org/259326@main
